tutorials/interactive/index.md

```diff
@@ -0,0 +1,814 @@
+          [[highlight]]new FavoriteWidget()[[/highlight]],
+        ],
+      ),
+    );
+
+    return new MaterialApp(
+      // ...
+    );
+  }
+}
+{% endprettify %}
+
+<br>That's it! When you hot reload the app, the star icon should now respond to taps.
+
+### Problems?
+
+If you can't get your code to run, look in your IDE for possible errors.
+[Debugging Flutter Apps](https://flutter.io/debugging/) might help.
+If you still can't find the problem,
+check your code against the interactive Lakes example on GitHub.
+
+* [`lib/main.dart`](https://raw.githubusercontent.com/flutter/website/master/_includes/_code/lakes-interactive/main.dart)
+* [`pubspec.yaml`](https://raw.githubusercontent.com/flutter/website/master/_includes/_code/lakes-interactive/pubspec.yaml)&mdash;no changes to this file
+* [`lakes.jpg`](https://github.com/flutter/website/blob/master/_includes/_code/lakes-interactive/images/lake.jpg)&mdash;no changes to this file
+
+Refer to [Get support](/#get-support) if you still have questions.
+
+---
+
+The rest of this page covers several ways a widget's state can be managed,
+and lists other available interactive widgets.
+
+<a name="managing-state"></a>
+## Managing state
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* There are different approaches for managing state.
+* You, as the widget designer, choose which approach to use.
+* If in doubt, start by managing state in the parent widget.
+
+</div>
+
+Who manage's the stateful widget's state? The widget itself? The parent widget?
+Both? Another object? The answer is... it depends.
+There are several valid ways to make your widget interactive.
+You, as the widget designer, makes the decision based on how you expect your
+widget to be used. Here are the most common ways to manage state:
+
+* [The widget manages its own state](#self-managed)
+* [The parent manages the widget's state](#parent-managed)
+* [A mix-and-match approach](#mix-and-match)
+
+{% comment %}
+NOTE: Commenting this out for now. The example needs some updates.
+
+First, fix TapboxD, add it back to the repo, and then restore this note.
+<aside class="alert alert-info" markdown="1">
+**Note:** You can also manage state by exporting the state to a model class
+that notifies widgets when state changes have occurred. This approach is
+particularly useful when you want multiple widgets to listen and respond to the
+same state information.
+
+Explaining this approach is beyond the scope of this tutorial,
+but you can try it out using the TapboxD example on GitHub.
+The only file you need is
+[lib/main.dart]().
+[PENDING: Add a link once it's up on the site.]
+</aside>
+{% endcomment %}
+
+How do you decide which approach to use?
+
+The general rule of thumb is that the parent widget should manage the
+state until it becomes too much of a burden. If there is any state
```

Comment:
Here's another rule of thumb which I think may be useful:

- If the state in question is user data, for example the checked/unchecked mode of a checkbox or the position of a slider, then the state is best managed by the parent widget.
- If the state in question is aesthetic, for example an animation, then the state is best managed by the widget itself.